### PR TITLE
Set test options via cabal.project

### PR DESCRIPTION
**Pull request by michaelpj**
This means we can set the default properly for people using the project, and also set things more simply in the test workflow.

The option setting in `hls-test-options` also seemed to be simply not working, for reasons I don't fully understand (but it seems like `NumThreads` is a special option.

### CODE REVIEW

**Review by fendor — APPROVED** [review]
LGTM

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -104,151 +104,147 @@ jobs:
...
-        # run the tests without parallelism, otherwise tasty will attempt to run
-        # all functional test cases simultaneously which causes way too many hls
-        # instances to be spun up for the poor github actions runner to handle
-        #
         # See https://github.com/ocharles/tasty-rerun/issues/22 for why we need
         # to include 'new' in the filters, since many of our test suites are in the
         # same package.
         run: |
-          echo "TEST_OPTS=-j1 --rerun-update --rerun-filter failures,exceptions,new" >> $GITHUB_ENV
+          echo 'test-options: --rerun-update --rerun-filter "failures,exceptions,new"' >> cabal.project.local
```
Comment (jhrcek) [inline .github/workflows/test.yml:111]: Nitpick: isn't this what [cabal configure](https://cabal.readthedocs.io/en/stable/cabal-commands.html#cabal-configure) is for?
e.g.
cabal configure --test-options='--rerun-update --rerun-filter failures,exceptions,new'

creates cabal.project.local with the desired options.

Comment (michaelpj) [reply .github/workflows/test.yml:111]: Good point!

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -104,151 +104,147 @@ jobs:
...
-        # run the tests without parallelism, otherwise tasty will attempt to run
-        # all functional test cases simultaneously which causes way too many hls
-        # instances to be spun up for the poor github actions runner to handle
-        #
         # See https://github.com/ocharles/tasty-rerun/issues/22 for why we need
         # to include 'new' in the filters, since many of our test suites are in the
         # same package.
         run: |
-          echo "TEST_OPTS=-j1 --rerun-update --rerun-filter failures,exceptions,new" >> $GITHUB_ENV
+          echo 'cabal configure --test-options="--rerun-update --rerun-filter failures,exceptions,new"' >> cabal.project.local
```
Comment (jhrcek) [inline .github/workflows/test.yml:111]: I mean this, which has the side effect of writing those options into cabal.project.local
```suggestion
          cabal configure --test-options="--rerun-update --rerun-filter failures,exceptions,new"
```

```
$ cabal configure --test-options="--rerun-update --rerun-filter failures,exceptions,new"
$ cat cabal.project.local 
ignore-project: False
test-options: "--rerun-update"
              "--rerun-filter"
              "failures,exceptions,new"
```

Comment (michaelpj) [reply .github/workflows/test.yml:111]: That's what I meant to type, just :facepalm:

**Pull request merged**